apis/python/src/tiledbsoma/_read_iters.py

```diff
@@ -63,8 +64,18 @@
 class TableReadIter(somacore.ReadIter[pa.Table]):
     """Iterator over `Arrow Table <https://arrow.apache.org/docs/python/generated/pyarrow.Table.html>`_ elements"""
 
-    def __init__(self, sr: clib.SOMAArray):
-        self._reader = _arrow_table_reader(sr)
+    def __init__(
+        self,
+        array: SOMAArray,
+        coords: options.SparseDFCoords,
```

Comment:
It's a little weird that on the one hand this takes a fully generic `SOMAArray`, but on the other hand, the coords are `options.Sparse_DFCoords` ... 🤔

Reply:
Maybe this makes total sense ... the types that can be passed to a `DenseNDArray` or `SparseNDArray` read are a strict subset of the types that can be passed to a `DataFrame` read ... maybe just add a comment saying that: this covers the union of all types that are legitimate for `coords` ...

Reply:
Good catch, ty. I've corrected this to `Union[options.SparseDFCoords, options.DenseNDCoords]`.

Reply:
Union also with `SparseNDCoords`?